### DISCUSSION TITLE
Want to check on Sidekiq's update policy

### DISCUSSION THREAD

**Discussion by S-H-GAMELINKS** [mastodon/mastodon#32668] [Q&A]
Hi
I recently updated to Sidekiq 7.3 (along with other related gems) on my Masotodon.
I am wondering if I can throw a PR to mastodon/mastodon as well, since it seemed to be working fine at the moment.

However, I did not know how the update to Sidekiq v7 was proceeding, so I opened Disscussion.

I would be happy to hear from you about your policy on updating Sidekiq.

For your information, here is a sample change diff to merge into mastodon/mastodon's main
https://github.com/S-H-GAMELINKS/mastodon/pull/1540

Should I submit a PR based on this one, etc.?

**Comment by StrangeMinds1960** [comment]
Can I ask how you upgraded Sidekiq?, because I've found two different ways to upgrade on the Sidekiq Github site

**Reply by S-H-GAMELINKS** [reply]
>Can I ask how you upgraded Sidekiq?

After upgrading the Sidekiq version in the Gemfile, I ran some tests to see if anything had changed in the configuration, fixed it, and updated it

> because I've found two different ways to upgrade on the Sidekiq Github site

Are you referring to the following documentation for this?
https://github.com/sidekiq/sidekiq/blob/main/docs/7.0-Upgrade.md
https://github.com/sidekiq/sidekiq/blob/main/docs/7.0-API-Migration.md

The basic differences are described in 7.0-Upgrade.md, and I've modified it to include getting Sidekiq configuration values, etc.
The changes to the Sidekiq API are described in 7.0-API-Migration.md, and the Sidekiq logger, etc. are modified based on this document.

**Reply by StrangeMinds1960** [reply]
I also saw on

https://github.com/sidekiq/sidekiq

Use bundle up sidekiq to upgrade Sidekiq and all its dependencies. Upgrade notes between each major version can be found in the docs/ directory.

(scroll down)

**Comment by mjankowski** [comment]
This is blocked somewhat on figuring out how to handle the namespace/DB portion of the upgrade. Background - https://github.com/mastodon/mastodon/pull/25988
